src/Airship/Internal/Route.hs

```diff
@@ -65,6 +102,42 @@ var t = Route [Var t]
 star :: Route
 star = Route [RestUnbound]
 
+
+-- Routing trie creation algorithm
+-- 1. Store full paths as keys up to first `var`
+-- 2. Take SHA256 digest of the URL portion preceding the
```

Comment:
Why are do we calculate the hash with SHA256? SHA256 is designed to be secure, and by nature of that property, it is not particularly fast. (see http://thepasswordproject.com/oclhashcat_benchmarking. SHA256 is more than 7x slower than MD5, for example)

I think what we want is for the hash to be _unique_, for which there are a lot faster non-secure algorithms. MurmurHash is generally a good one, but would require another dependency. `hashable` might work, though I know you need the `ByteString` for the Trie. I'll take a look and give you a suggestion.

Otherwise, MD4 (`Crypto.Hash.MD4`) is pretty fast.